server/Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM node:17.3.0-alpine AS development
+
+RUN mkdir -p /usr/app/server
+WORKDIR /usr/app/server
+
+COPY package.json .
+
+RUN npm install
+
+COPY . .
+
+RUN npm run build
+
+FROM node:17.3.0-alpine as production
+
+ARG NODE_ENV=production
+ENV NODE_ENV=${NODE_ENV}
+
+WORKDIR /usr/app/server
+
+COPY package.json .
```

Comment:
better use lock files, or you can face some problems to receive reproducible builds.